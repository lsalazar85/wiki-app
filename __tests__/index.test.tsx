/**
 * @jest-environment jsdom
 */

import { MockedProvider } from "@apollo/react-testing";
import {render, screen } from '@testing-library/react'
import Home from "../pages";
import { mocks } from "../__mocks__/mocks";

describe('Home', () => {
  it('should render successfully', () => {
    const { asFragment } = render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Home />
        </MockedProvider>
    )

    expect(asFragment()).toMatchSnapshot();
    expect(asFragment()).toBeDefined()
  })

  it('should first render buttons elements successfully ', () => {
    render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Home />
        </MockedProvider>
    );

    const clearBtn = screen.getByRole('button', { name: /clear search/i })
    const nextBtn = screen.getByRole('button', { name: /next/i })
    const prevBtn = screen.getByRole('button', { name: /prev/i })

    expect(clearBtn).toBeInTheDocument()
    expect(nextBtn).toBeInTheDocument()
    expect(prevBtn).toBeInTheDocument()
  })
})
