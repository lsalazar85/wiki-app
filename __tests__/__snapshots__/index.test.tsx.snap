// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render successfully 1`] = `
<DocumentFragment>
  <main
    class="styles__Main-sc-1d2ydxg-0 kdKjvQ"
  >
    <header
      class="styles__HeaderContainer-sc-1j9m33g-0 bsuzQp"
    >
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjcwIiBoZWlnaHQ9IjkwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <img
          alt="Rick and Morty Logo"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </div>
    </header>
    <form
      class="styles__SearchContainer-sc-1xiurla-0 cSNHGB"
    >
      <input
        class="styles__SearchInput-sc-1xiurla-1 dQGEEo"
        data-testid="search"
        placeholder="Search character by name"
        type="text"
        value=""
      />
    </form>
    <button
      class="styles__ButtonContainer-sc-9v32al-0 kRkStR"
    >
      Clear Search
    </button>
    <div
      class="styles__PaginationContainer-sc-1pxi0jf-0 hyggEk"
    >
      <div
        class="styles__PaginationDiv-sc-1pxi0jf-1 dSpzUZ"
      >
        <button
          class="styles__ButtonContainer-sc-9v32al-0 kRkStR"
        >
          Prev
        </button>
        <span
          class="styles__PaginationText-sc-1pxi0jf-2 eSGdAL"
        >
           of 
        </span>
        <button
          class="styles__ButtonContainer-sc-9v32al-0 bRlmzH"
          disabled=""
        >
          Next
        </button>
      </div>
    </div>
    <section
      class="styles__ContentContainer-sc-773toi-0 dKzdKd"
    >
      <div
        class="styles__LoaderContainer-sc-1sn62nv-0 btuLRX"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iOTAiIGhlaWdodD0iOTAiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgdmVyc2lvbj0iMS4xIi8+"
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="Loader Icon"
            class="styles__LoaderImage-sc-1sn62nv-1 fwBpdA"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </div>
      </div>
      <div
        class="styles__CardSection-sc-1d2ydxg-1 kIIQyV"
      />
    </section>
  </main>
</DocumentFragment>
`;
